app/api/margatsni/v1/endpoints/posts.rb

```diff
@@ -39,11 +40,15 @@ def represent_post(post)
           desc 'Create new post'
           params do
             requires :body, type: String, length: 500
+            requires :image_attributes, type: Hash do
+              requires :file_data, type: File, allow_blank: false
+            end
           end
           post do
-            post = current_user.posts.create(declared(params))
+            post = current_user.posts.build(declared(params, include_missing: false))
+            post.save
```

Comment:
You forgot about validations. In case if the image will be too large, your record will not be saved and you would need to process record errors.
something like 

```
if post.save
  represent_post(post.reload)
else
  present_errors_for(post)
end
```

and helper
```
def present_errors_for(entity)
  # some reusable code to process validation errors
end
```